cli/lib/gather.js

```diff
@@ -197,7 +197,7 @@ class Walker {
 			return;
 		}
 		const info = new FileInfo(name, from, to);
-		const relPath = from.replace((origSrc || src) + path.sep, prefix ? prefix + path.sep : '').replace(/\\/g, '/');
+		const relPath = from.replace((origSrc || src) + path.sep, prefix ? prefix + path.sep : '').replace(/\\/g, path.sep);
```

Comment:
Ah, I see the issue. Ok, so I had assumed this would normalize paths to unix style path separators - but the later calls when we walk directories ends up using `path.join()` which uses the OS-path separator. The `path`/`fs` APIs will still handle using either OK, but the regex matching obviously wouldn't work as expected (and there's potential for duplicated paths ending up in the map/sets as separate entries even though they meant to point to the same file but may have mismatching separators.)

Reply:
So the downstream effects here would likely fall to any of the "tasks" that end up processing the files in these sets/groups of files. So the places to look would be:
- https://github.com/appcelerator/titanium_mobile/tree/master/cli/lib/tasks
- https://github.com/appcelerator/titanium_mobile/blob/master/android/cli/lib/process-drawables-task.js
- https://github.com/appcelerator/titanium_mobile/blob/master/android/cli/lib/process-splashes-task.js

Of those, I think this is the only line that needs to be changed (though technically it's iOS-only so `path.sep` should always be `'/'` on the OS): https://github.com/appcelerator/titanium_mobile/blob/master/cli/lib/tasks/process-js-task.js#L268

The other thing to double-check/confirm is that the underlying code in `apps-tasks` will end up recording hr files using the proper `path.sep` so that the caching/keys it used matched what we used. I think we are the ones who report the directories/filepaths to it, so as long as we're consistent I think we're fine.